### PR TITLE
ATM QoL and bug fixes

### PR DESCRIPTION
**Pull request by MadmanMartian**
After seeing 
image in the thread, I thought I'd take a shot at fixing this.

 - If you're wearing a PDA, cash from the ATM is put into your virtual wallet (Might disable this, as there's the "transfer to virtual wallet" option)
 - It will then try to put the money on the turf "behind" itself
 - If it can not reach said turf, it'll just eject the money with force.

 - ~~Also adds a button for toggling the ATMs ability to take money from the floor~~ Removes ATMs taking money from the floor, as that was a bodge-job fix for PDAs printing money and putting it right on the floor.

 - Makes adding money to your virtual wallet into a function, currently used by applying cash to your PDA, and by the ATM placing money in your virtual wallet

:cl:
 * rscadd: ATMs now check if the user can actually reach the turf they are going to place the money they are removing from their account onto, otherwise it'll just throw the money at them.
 * rscadd: ATMs no longer steal money from the floor
 * tweak: Rather than printing money from your PDA putting the money on the floor, instead it attempts to put the money in your hands whenever possible.

### CODE REVIEW

**Comment by SonixApache** [comment]
I sincerely wish you would just get the money in your hands.

**Comment by MadmanMartian** [comment]
It's not always just one piece, but sure I'll see what I can do

**Comment by jknpj** [comment]
Some kind of container to hold the money so you can drag the ATM to you and grab your cash, like a real ATM

**Comment by Tehsapper** [comment]
being able to rob ATMs when?

**Comment by Kurfursten** [comment]
Note you do not need a PDA to access your virtual wallet, it's in the ID

File: code/WorkInProgress/Mini/ATM.dm [hunk 1]
```diff
@@ -550,7 +554,21 @@ log transactions
 			dispense_cash(arbitrary_sum,H.wear_id)
 			to_chat(usr, "[bicon(src)]<span class='notice'>Funds were transferred into your physical wallet!</span>")
 			return
-	dispense_cash(arbitrary_sum,get_step(get_turf(src),turn(dir,180))) // Spawn on the ATM.
+		if(istype(H.wear_id, /obj/item/device/pda))
+			var/obj/item/device/pda/P = H.wear_id
```
Comment (Kurfursten) [inline code/WorkInProgress/Mini/ATM.dm:536]: this is what I was talking about

File: code/WorkInProgress/Mini/ATM.dm [hunk 2]
```diff
@@ -550,7 +554,21 @@ log transactions
...
+			to_chat(usr, "[bicon(src)]<span class='notice'>Funds were transferred into your virtual wallet!</span>")
+			P.add_to_virtual_wallet(arbitrary_sum, user)
+			return
+	var/turf/our_turf = get_turf(src)
+	var/turf/destination_turf = get_step(our_turf, turn(dir, 180))
```
Comment (Kurfursten) [inline code/WorkInProgress/Mini/ATM.dm:546]: would it not be more reliable to just step toward user?

**Comment by Kurfursten** [comment]
today I will remind him

**Comment by Kurfursten** [comment]
This doesn't need a lot of work to be finished

File: code/WorkInProgress/Mini/ATM.dm (hunk 2 above)
Comment (MadmanMartian) [reply code/WorkInProgress/Mini/ATM.dm:546]: Stepping towards user may put it in the "consume" turf

Comment (MadmanMartian) [reply code/WorkInProgress/Mini/ATM.dm:546]: I've removed the consume turf stuff

**Pull request merged**